hack/lib/util/golang.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+#
+# This library holds golang related utility functions.
+
+# os::golang::verify_go_version ensure the go tool exists and is a viable version.
+function os::golang::verify_go_version() {
+	if ! which go &>/dev/null; then
+		os::log::error "Can't find 'go' in PATH, please fix and retry."
+		os::log::error "See http://golang.org/doc/install for installation instructions."
+		return 1
+	fi
+
+	local go_version
+	go_version=($(go version))
+	if [[ "${go_version[2]}" != "go1.6" && -z "${PERMISSIVE_GO:-}" ]]; then
+		os::log::error "Detected go version: ${go_version[*]}."
+		os::log::error "Origin requires go version 1.6."
+		os::log::error "Please install Go version 1.6 or use PERMISSIVE_GO=y to bypass this check."
```

Comment:
Specifically it is very important that if someone is using this flag they are aware that there is a very real chance that everything could look great locally but the stuff they see on the CI will not be the same.